py_ecc/bls/point_compression.py

```diff
@@ -30,6 +30,30 @@
 )
 
 
+#
+# The most-significant three bits of a G1 or G2 encoding should be masked away before
+# the coordinate(s) are interpreted.
+# These bits are used to unambiguously represent the underlying element
+# The format: (c_flag, b_flag, a_flag, x)
+# https://github.com/zcash/librustzcash/blob/6e0364cd42a2b3d2b958a54771ef51a8db79dd29/pairing/src/bls12_381/README.md#bls12-381-instantiation  # noqa: E501
+#
+def get_flags(z: int) -> Tuple[bool, bool, bool]:
+    c_flag = bool((z >> 383) & 1)  # The most significant bit.
+    b_flag = bool((z >> 382) & 1)  # The second-most significant bit.
+    a_flag = bool((z >> 381) & 1)  # The third-most significant bit.
+    return c_flag, b_flag, a_flag
+
+
+def is_point_at_infinity(z1: int, z2: Optional[int] = None) -> bool:
+    """
+    If z2 is None, the given z1 is a G1 point.
+    Else, (z1, z2) is a G2 point.
+    """
+    return (z1 % POW_2_381 == 0) and (
+        z2 is None or (z2 is not None and z2 == 0)
+    )
```

Comment:
Looks like equivalent to
```suggestion
    return (z1 % POW_2_381 == 0) and (z2 is None or z2 == 0)
```